server/database.sql

```diff
@@ -1,150 +1,378 @@
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `contests_moderators`
+--
+
+DROP TABLE IF EXISTS `contests_moderators`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `contests_moderators` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `contest_id` int NOT NULL,
+  `moderator` varchar(45) NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`),
+  KEY `fk_contests_moderators_1_idx` (`contest_id`),
+  KEY `fk_contests_moderators_2_idx` (`moderator`),
+  CONSTRAINT `fk_contests_moderators_1` FOREIGN KEY (`contest_id`) REFERENCES `contests` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `fk_contests_moderators_2` FOREIGN KEY (`moderator`) REFERENCES `users` (`username`) ON DELETE RESTRICT ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `contests_participants`
+--
+
+DROP TABLE IF EXISTS `contests_participants`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `contests_participants` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `contest_id` int NOT NULL,
+  `participant` varchar(45) NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`),
+  KEY `fk_contests_participants_1_idx` (`contest_id`),
+  KEY `fk_contests_participants_2_idx` (`participant`),
+  CONSTRAINT `fk_contests_participants_1` FOREIGN KEY (`contest_id`) REFERENCES `contests` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `fk_contests_participants_2` FOREIGN KEY (`participant`) REFERENCES `users` (`username`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `contests_questions`
+--
+
+DROP TABLE IF EXISTS `contests_questions`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `contests_questions` (
+  `id` int NOT NULL AUTO_INCREMENT,
+  `question_id` int NOT NULL,
+  `contest_id` int NOT NULL,
+  `max_score` int NOT NULL,
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `id_UNIQUE` (`id`),
+  KEY `fk_contests_questions_1_idx` (`contest_id`),
+  KEY `fk_contests_questions_2_idx` (`question_id`),
+  CONSTRAINT `fk_contests_questions_1` FOREIGN KEY (`contest_id`) REFERENCES `contests` (`id`) ON DELETE CASCADE ON UPDATE CASCADE,
+  CONSTRAINT `fk_contests_questions_2` FOREIGN KEY (`question_id`) REFERENCES `questions` (`id`) ON DELETE CASCADE ON UPDATE CASCADE
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
```

Comment:
Similarly, 1 more constraint needs to be added to avoid unnecessary duplicate entries -
```
UNIQUE KEY `unique_contests_questions` (`contest_id`,`question_id`)
```